### ISSUE TITLE
Feature request: precision & recall

### ISSUE THREAD

**Issue by ghost** [softwaredeng/inTrees#12]
My data set is imbalanced; I understand from the [Interpreting Tree Ensembles with inTrees](https://arxiv.org/pdf/1408.5456v1.pdf) that Error = accuracy. 

In an imbalanced set Error rate is not very useful. If you have 95% classA and 5% classB then the accuracy is 95% if you predict classA regardless. Many datasets are imbalanced. Therefore recall and precision would be better indicators of the rules quality.

**Comment by GillesVandewiele** [comment]
You're completely right that making the error metric a parameter of the algorithm would indeed be an improvement. I think implementing this yourself would not be too hard by adapting the the [measureRule.R](https://github.com/softwaredeng/inTrees/blob/42b792deefc003183ac518f23d9d9de1ec11cc33/inTrees/R/measureRule.R) file.

Also, better to optimize F1 than either recall or precision (since making a classifier that is able to achieve a precision or recall of 1.0 is straight-forward; e.g. if you use recall (or 1 - recall) as your error metric, you will just get a model that always predicts the positive class (since for that, the recall = 1.0))

**Comment by ghost** [comment]
Yes, that makes sense. I like to see the precision and recall as it depends on the application which one you would like to maximize. 

The `measureRule.R` is an internal function - where does it fit in? The example in `buildLearner` features an intermediate result `ruleExec` (both raw and unique) is that the same `ruleExac` as the input var in `measureRule.R`?

**Comment by GillesVandewiele** [comment]
Unfortunately, I'm not too familiar with the inTrees internals, leave alone the language R...

Do you just want to know the precision and recall of your final model? Else, you can just measure it after training the model

Or, do you want to optimize your model such that it maximizes the F1-score? If that's the case, the internals will need to be changed. I just took a quick glance and figured `measureRule` was the right place to do it...

**Comment by ghost** [comment]
Yes, I can calculate recall and precision of individual rules, but that's an end-of-pipe solution. measureRule looks like the right place, but I would need to know how it fits into the overall structure - that's a documentation issue. Let's close then.
